src/UID2.Client/UID2Encryption.cs

```diff
@@ -508,21 +533,25 @@ private static byte[] GenerateIV(int len = 16)
             return iv;
         }
 
-        private static IdentityScope DecodeIdentityScopeV3(byte value)
+        private static IdentityScope DecodeEncryptedDataIdentityScopeV3(byte prefix)
+        {
+            return (IdentityScope)((prefix >> 4) & 1);
+        }
+
+        private static bool ValidIdentityScopeAndTypeV3(byte prefix)
         {
-            return (IdentityScope)((value >> 4) & 1);
+            return TryDecodeIdentityScopeAndTypeV3(prefix, out _, out _);
         }
 
-        private static IdentityType GetIdentityType(byte[] encryptedId)
+        private static bool TryDecodeIdentityScopeAndTypeV3(byte prefix, out IdentityScope scope, out IdentityType type)
         {
-            // For specifics about the bitwise logic, check:
-            // Confluence - UID2-79 UID2 Token v3/v4 and Raw UID2 format v3
-            // In the base64-encoded version of encryptedId, the first character is always either A/B/E/F.
-            // After converting to binary and performing the AND operation against 1100,the result is always 0X00.
-            // So just bitshift right twice to get 000X, which results in either 0 or 1.
-            byte idType = encryptedId[0];
-            byte piiType = (byte)((idType & 0b_1100) >> 2);
-            return piiType == 0 ? IdentityType.Email : IdentityType.Phone;
+            // 4th bit == 1 then EUID otherwise UID2
+            scope = (IdentityScope) (prefix >> 4);
+            // 2nd bit == 1 then Phone else Email
+            type = (IdentityType) ((prefix >> 2) & 0b11);
+
+            return Enum.IsDefined(typeof(IdentityScope), scope) && Enum.IsDefined(typeof(IdentityType), type) && 
+                ((prefix & 0b11) == 0b11 || (prefix & 0b11) == 0);
```

Comment:
Can we add a comment that prefix should be 0b11, but the current Python SDK uses 0b00 (that's why we're allowing it, right?)

Reply:
why do we need to have special python SDK handling code when we didn't before?

Reply:
We didn't validate the 1st and 0th bit before, but now we are

Reply:
What's the downside of not validating these bits? Can we still get a good error by looking at other things like length, version in an acceptable range, etc?